torch2jax/__init__.py

```diff
@@ -211,6 +218,35 @@ def _coerce(x):
     )
 
 
+def _tree_coerce(x):
+  spec = torch_tree_structure(x)
+  specs = deque([spec])
+
+  while len(specs) > 0:
+    spec = specs.popleft()
+    specs.extend(spec.children_specs)
+    if spec.is_leaf():
+      continue
+    if spec.type in jax._src.tree_util._registry:
+      continue  # already registered
+    node = torch.utils._pytree.SUPPORTED_NODES.get(spec.type, None)
+    if node is None:
+      continue
+    if is_dataclass(spec.type):
+      jax.tree_util.register_dataclass(spec.type)
+    else:
+      jax.tree_util.register_pytree_node(
+        spec.type,
+        node.flatten_fn,
+        lambda a, b, n=node: n.unflatten_fn(b, a),  # torch & jax have different order of args
+        # flatten_with_keys_func=node.flatten_with_keys_fn,
+        # TODO: flatten_with_keys has some problem as jax & torch has different class for key
+        # without registering it seems to just work as well.
```

Comment:
```suggestion
```
flatten_with_keys_func is undocumented in jax and not referenced in any of the examples. i say we just skip it entirely

Reply:
Sounds good